### PR TITLE
Fixed BodyOrientation refactor regressions

### PR DESCRIPTION
**Pull request by Mailaender**
Another followup of https://github.com/OpenRA/OpenRA/pull/9004 which fixes problems discovered by Coverity and myself.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
I get "Actor type 'camera.paradrop' does not define a quantized body orientation." when using paratroopers with the PR, just like I got with #8839. That's why I removed the sync check in mine.

**Comment by Mailaender** [comment]
Fixed.

File: OpenRA.Mods.Common/Traits/BodyOrientation.cs [hunk 1]
```diff
@@ -76,9 +76,14 @@ public BodyOrientation(ActorInitializer init, BodyOrientationInfo info)
 				var isb = self.HasTrait<ISpriteBody>();
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/BodyOrientation.cs]: Looks like you can move this after the ```qboi == null``` check and avoid the lookup most of the time.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/BodyOrientation.cs]: Good idea. Done.

**Comment by penev92** [comment]
:+1:

File: OpenRA.Mods.Common/Traits/BodyOrientation.cs [hunk 2]
```diff
@@ -73,12 +73,16 @@ public BodyOrientation(ActorInitializer init, BodyOrientationInfo info)
...
-				var isb = self.HasTrait<ISpriteBody>();
 
 				// If a sprite actor has neither custom QuantizedFacings nor a trait implementing IQuantizeBodyOrientationInfo, throw
-				if (qboi == null && isb)
-					throw new InvalidOperationException("Actor" + self.Info.Name + "has a sprite body but no facing quantization."
-						+ " Either add the QuantizeFacingsFromSequence trait or set custom QuantizedFacings on BodyOrientation.");
+				if (qboi == null)
+				{
+					if (self.HasTrait<ISpriteBody>())
+						throw new InvalidOperationException("Actor" + self.Info.Name + "has a sprite body but no facing quantization."
```
Comment (penev92) [inline OpenRA.Mods.Common/Traits/BodyOrientation.cs]: Please change the beginning to 
>("Actor " + self.Info.Name + "`` has a ...

for easier readability. Right now this results in "Actorlpst2has a sprite body".

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/BodyOrientation.cs]: Right. Done.

**Comment by reaperrr** [comment]
:+1:

**Pull request merged**